### DISCUSSION TITLE
Problem uploading a translation through a nuget package

### DISCUSSION THREAD

**Discussion by elgransan** [oqtane/oqtane.framework#2190] [General]
Hi, I recently made a PR with Spanish translation for the v3.1.1

I've generated the nuget package to upload the language pack to my instance, all works ok but when I select the spanish language on the selector, It keeps the english texts

This is the nuget package I'm uploading to my instance (with JetBrains dotPeek)
![image](https://user-images.githubusercontent.com/9949584/167445474-a04e1124-3d6d-42cd-8b70-be4f0648f1e7.png)

Spanish selection still in english
![image](https://user-images.githubusercontent.com/9949584/167445983-88db46ba-5544-4e38-990b-a1cc09306440.png)

I can see the file on the project
![image](https://user-images.githubusercontent.com/9949584/167446118-190ec95d-d8cc-4ed6-b0a8-cf01bceda5e4.png)

Any idea what is wrong?

**Comment by elgransan** [comment]
The only difference comparing to netherland localization I see is that I use the generic language culture instead country culture (in version 2 that wasn't a problem)

And the culture is changing, I got the page in spanish language and I'm using Humanizer component and it's getting the spanish version too

**Reply by elgransan** [reply]
Can someone help me to test it, I uploaded the package on NuGet.org [Oqtane.Client.es](https://www.nuget.org/packages/Oqtane.Client.es/), I'd appreciate it very much

**Reply by sbwalker** [reply]
@elgransan I did some research into this today and I am unable to determine why your ES language resources are not working. When I stepped through the code I could see that the satellite DLL was being located in the appropriate location and the DLL was loaded into the app domain. The localization cookie also seems to be set correctly when I choose the Spanish option from the language selector in the UI. However the IStringLocalizer (which is part of Microsoft.Extensions.Localization) is not returning the Spanish value for a specific resource key - it is returning the English value (ie. the fallback). Perhaps @leigh-pointer has some ideas...

**Reply by elgransan** [reply]
Thank you @sbwalker I'm going to change all the filenames with "es" language for "es-ES" to rule out that case.

**Reply by elgransan** [reply]
Ok, I could solve the problem, It was the Resource namespace, must be "Oqtane.Resources" instead of "Oqtane.Client.Resources"

![image](https://user-images.githubusercontent.com/9949584/169635297-08935135-1815-484b-91b8-9bf4953455d2.png)

Setting this at the project solve the problem
![image](https://user-images.githubusercontent.com/9949584/169635319-597c9525-0caa-428f-a266-e2e5877ec557.png)

**Reply by sbwalker** [reply]
@elgransan I am very glad you were able to resolve the issue
